### PR TITLE
Fixes #34239: Add pull-mqtt support to smart_proxy_remote_execution

### PR DESCRIPTION
**Pull request by ehelms**
Builds on:

- https://github.com/theforeman/puppet-foreman_proxy/pull/736
- https://github.com/theforeman/puppet-foreman_proxy/pull/735
- https://github.com/theforeman/puppet-foreman_proxy/pull/734

### CODE REVIEW

**Comment by evgeni** [comment]
given all this builds upon is merged, I've rebased it and made the linter happy ;)

File: manifests/plugin/remote_execution/script.pp [hunk 1]
```diff
@@ -70,4 +70,61 @@
...
+        "keyfile ${mosquitto_ssl_dir}/ssl_key.pem",
+        'require_certificate true',
+        'use_identity_as_username true',
+      ],
+      require      => File['/etc/mosquitto/foreman.acl'],
+    }
+
+    file { "${mosquitto_config_dir}/foreman.acl":
+      ensure  => 'file',
+      content => epp("${module_name}/plugin/foreman.acl.epp", { 'user' => $facts['networking']['fqdn'] }),
```
Comment (evgeni) [inline manifests/plugin/remote_execution/script.pp:95]: One thing I can never remember, are our certs lower-case or not?

We have this explicit `lower_fqdn` variable:
https://github.com/theforeman/puppet-foreman_proxy/blob/bd47afb7d0b4d6042c361c2d9238543ea8f1c240/manifests/params.pp#L4

**Comment by evgeni** [comment]
the test failures are legit ;)

File: manifests/plugin/remote_execution/mosquitto.pp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    mode    => '0640',
+    require => Package['mosquitto'],
+  }
+
+  file { $mosquitto_ssl_dir:
+    ensure  => directory,
+    owner   => 'root',
+    group   => 'mosquitto',
+    mode    => '0755',
+    require => Package['mosquitto'],
```
Comment (evgeni) [inline manifests/plugin/remote_execution/mosquitto.pp:70]: shouldn't this get an autorequire on `File[$mosquitto_config_dir]` anyways?

File: manifests/plugin/remote_execution/mosquitto.pp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  file { "${mosquitto_ssl_dir}/ssl_cert.pem":
+    ensure  => 'file',
+    content => file($ssl_cert),
+    owner   => 'root',
+    group   => 'mosquitto',
+    mode    => '0440',
+    require => Package['mosquitto'],
```
Comment (evgeni) [inline manifests/plugin/remote_execution/mosquitto.pp:79]: shouldn't this get an autorequire on `File[$mosquitto_ssl_dir]` anyways?

File: manifests/plugin/remote_execution/mosquitto.pp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  file { "${mosquitto_ssl_dir}/ssl_key.pem":
+    ensure  => 'file',
+    content => file($ssl_key),
+    owner   => 'root',
+    group   => 'mosquitto',
+    mode    => '0440',
+    require => Package['mosquitto'],
```
Comment (evgeni) [inline manifests/plugin/remote_execution/mosquitto.pp:88]: :arrow_up:

File: manifests/plugin/remote_execution/mosquitto.pp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  file { "${mosquitto_ssl_dir}/ssl_ca.pem":
+    ensure  => 'file',
+    content => file($ssl_ca),
+    owner   => 'root',
+    group   => 'mosquitto',
+    mode    => '0440',
+    require => Package['mosquitto'],
```
Comment (evgeni) [inline manifests/plugin/remote_execution/mosquitto.pp:97]: :arrow_up:

**Comment by evgeni** [comment]
Should mosquitto be cleaned up if the user decides to flip back to SSH?

**Review by evgeni — APPROVED** [review]
I think this addresses all points we had discussed over in #726.

Should this PR/commit ref/fix RM#34239? I think it should.

We can take the "switch from mqtt to ssh" question to another day/PR.

**Comment by ehelms** [comment]
> We can take the "switch from mqtt to ssh" question to another day/PR.

I started a separate PR dedicated to that -- https://github.com/theforeman/puppet-foreman_proxy/pull/744 . I will work on the hard part -- tests to verify it works.

**Comment by ehelms** [comment]
@adamruzicka do you want to take a look or do any testing before merge?

**Comment by ehelms** [comment]
I nearly forgot we need to add a hard dependency in the installer given puppet-mosquitto is a soft dependency in the module: https://github.com/theforeman/foreman-installer/pull/760

**Pull request merged**